pkg/volume/flexvolume/plugin.go

```diff
@@ -122,21 +122,14 @@ func (plugin *flexVolumePlugin) GetVolumeName(spec *volume.Spec) (string, error)
 	call := plugin.NewDriverCall(getVolumeNameCmd)
 	call.AppendSpec(spec, plugin.host, nil)
 
-	_, err := call.Run()
+	resp, err := call.Run()
 	if isCmdNotSupportedErr(err) {
 		return (*pluginDefaults)(plugin).GetVolumeName(spec)
 	} else if err != nil {
 		return "", err
 	}
 
-	name, err := (*pluginDefaults)(plugin).GetVolumeName(spec)
```

Comment:
Please use the API and reduce the verbosity of the call. Thanks

Reply:
Can you expand on that @chakri-nelluri ? It looks like we never actually returned volume name returned by the flex volume call. That sounds like a bug.

Reply:
History :) The volume name was supposed to be unique which was tough to be enforced on the driver side and there is no clear mapping from driver volume name to PV name. To avoid this confusion we always defaulted to default volume name.

Reply:
@chakri-nelluri Thanks for your clarification.